src/fswAlgorithms/imageProcessing/centerOfBrightness/centerOfBrightness.cpp

```diff
@@ -0,0 +1,131 @@
+ */
+void CenterOfBrightness::Reset(uint64_t CurrentSimNanos)
+{
+    // check that the required message has not been connected
+    if (!this->imageInMsg.isLinked()) {
+        bskLogger.bskLog(BSK_ERROR, "CenterOfBrightness.imageInMsg wasn't connected.");
+    }
+}
+
+/*! This module reads an OpNav image and extracts the weighted center of brightness. It performs a greyscale, a blurr,
+ * and a threshold on the image before summing the weighted pixel intensities in order to average them with the
+ * total detected intensity. This provides the center of brightness measurement (as well as the total number of
+ * bright pixels)
+ @return void
+ @param CurrentSimNanos The clock time at which the function was called (nanoseconds)
+ */
+void CenterOfBrightness::UpdateState(uint64_t CurrentSimNanos)
+{
+    std::string dirName;
+    CameraImageMsgPayload imageBuffer;
+    OpNavCOBMsgPayload cobBuffer;
+
+    imageBuffer = this->imageInMsg.zeroMsgPayload;
+    cobBuffer = this->opnavCOBOutMsg.zeroMsgPayload;
+
+    cv::Mat imageCV, blurred, imageGray;
+    if (this->saveDir != ""){
+        dirName = this->saveDir + std::to_string(CurrentSimNanos*1E-9) + ".jpg";
+    }
+    else{dirName = "./"+ std::to_string(CurrentSimNanos*1E-9) + ".jpg";}
+    
+    /*! - Read in the image*/
+    if(this->imageInMsg.isLinked())
+    {
+        imageBuffer = this->imageInMsg();
+        this->sensorTimeTag = this->imageInMsg.timeWritten();
+    }
+    /* Added for debugging purposes*/
+    if (!this->filename.empty()){
+        imageCV = cv::imread(this->filename, cv::IMREAD_COLOR);
+    }
+    else if(imageBuffer.valid == 1 && imageBuffer.timeTag >= CurrentSimNanos){
+        /*! - Recast image pointer to CV type*/
+        std::vector<unsigned char> vectorBuffer((char*)imageBuffer.imagePointer, (char*)imageBuffer.imagePointer + imageBuffer.imageBufferLength);
+        imageCV = cv::imdecode(vectorBuffer, cv::IMREAD_COLOR);
+        if (this->saveImages == 1){
+            if (!cv::imwrite(dirName, imageCV)) {
+                bskLogger.bskLog(BSK_WARNING, "CenterOfBrightness: wasn't able to save images.");
+            }
+        }
+    }
+    else{
+        /*! - If no image is present, write zeros in message */
+        this->opnavCOBOutMsg.write(&cobBuffer, this->moduleID, CurrentSimNanos);
+        return;
+    }
+    cv::cvtColor( imageCV, imageGray, cv::COLOR_BGR2GRAY);
+    cv::blur(imageGray, blurred, cv::Size(this->blurrSize,this->blurrSize) );
+    cv::threshold(blurred, imageCV, this->threshold, 255, cv::THRESH_BINARY);
+    
+    std::vector<cv::Vec2i> locations;
+    /*! - Find all the non-zero pixels in the image*/
+    cv::findNonZero(imageCV, locations);
+    
+    cobBuffer.timeTag = this->sensorTimeTag;
+    cobBuffer.cameraID = imageBuffer.cameraID;
+    uint32_t weight, weightSum;
+    weightSum = 0;
+    /*!- If no lit pixels are found do not validate the image as a measurement */
+    if ((int) locations.size() > 0){
```

Comment:
:small_red_triangle_down: This can be changed to `!locations.empty()`

Reply:
Done